vermouth/processors/apply_rubber_band.py

```diff
@@ -129,35 +164,69 @@ def build_connectivity_matrix(graph, separation, selection=None):
     if separation < 0:
         raise ValueError('Separation has to be null or positive.')
     if separation == 0:
-        # The connectivity matrix with a separation of 1 is the adjacency
-        # matrix. Thanksfully, networkx can directly give it to us a a numpy
-        # array.
-        return nx.to_numpy_matrix(graph, nodelist=selection).astype(bool)
-    subgraph = graph.subgraph(selection)
-    connectivity = np.zeros((len(subgraph), len(subgraph)), dtype=bool)
-    for (idx, key_idx), (jdx, key_jdx) in itertools.combinations(enumerate(subgraph.nodes), 2):
-        try:
-            shortest_path = len(nx.shortest_path(subgraph, key_idx, key_jdx))
-        except nx.NetworkXNoPath:
-            # There is no path between key_i and key_j so they are not
-            # connected; which is the default.
-            pass
-        else:
-            # The source and the target are counted in the shortest path
-            connectivity[idx, jdx] = shortest_path <= separation + 2
-            connectivity[jdx, idx] = connectivity[idx, jdx]
-    return connectivity
+        # The connectivity matrix with a separation of 0 is the adjacency
+        # matrix. Thankfully, networkx can directly give it to us a a numpy
+        # matrix.
+        return np.asarray(nx.to_numpy_matrix(graph, nodelist=selection).astype(bool))
+    if selection is None:
+        selection = slice(None, None, None)
+    size = graph.number_of_nodes()
+    correspondence = {node: index for index, node in enumerate(graph.nodes)}
+    connectivity = np.zeros((size, size), dtype=bool)
+    # separation is provided in term of nodes while the path is provided in
+    # terms of edges, hence `separation + 1`.
+    distance_pairs = nx.all_pairs_shortest_path_length(graph, cutoff=separation + 1)
```

Comment:
Note that this will only work for 1-bead-per-backbone models. Maybe run it through graph_utils.make_residue_graph?

Reply:
It depends if the separation is taken as per particle or per residue. Both can be valid.

Reply:
True. How do you want to define it? :)
Should be documented somewhere user-facing though.

Reply:
@Tsjerk @jbarnoud how did martiniz1 deal with this? What was the default?

Reply:
Martinize 1 would not be able to have more than 1 backbone bead.

Reply:
@jbarnoud regarding this, you've already created some files for 2 BB bead protein models. Could some of those for testing? Otherwise I think this discussion is useless until someone wants to do a 2BB bead protein. So I'd follow Peter and just write it in the documentation somewhere. I think at this stage a CLI argument is not needed. We already have a ton for the EN.

Reply:
@fgrunewald GitHub does not let me attach my file so I sent it to you by email. You could look at DNA though.

Reply:
The mail did not go through. Here is a link https://www.dropbox.com/s/ivkjggnlapacze5/martini30bb.tar.gz?dl=0

Reply:
Thanks! I will give it a try and see how it goes. Probably we should also look at DNA again but I don't have a good reference yet. Let's see.

Reply:
multi back-bone bead models are working now, although the default is still name BB